main.go

```diff
@@ -33,33 +35,54 @@ func (c config) String() string {
 func main() {
 	var cfg config
 	envconfig.MustProcess("", &cfg)
-	fmt.Printf("Config: %s\n", cfg)
+	log.Printf("Config: %s\n", cfg)
 
 	client := slack.New(cfg.Token)
 
+	if cfg.UpdateTs != "" && cfg.DeleteTs != "" {
+		log.Fatal("Cannot update and delete a message at the same time")
+	}
+
 	// Send the message
 	options := []slack.MsgOption{content(cfg)}
-	if cfg.ThreadTs != "" {
+	if cfg.UpdateTs != "" {
+		options = append(options, slack.MsgOptionUpdate(cfg.UpdateTs))
+	} else if cfg.DeleteTs != "" {
+		options = append(options, slack.MsgOptionDelete(cfg.DeleteTs))
+	} else if cfg.ThreadTs != "" {
 		options = append(options, slack.MsgOptionTS(cfg.ThreadTs))
 		if cfg.AlsoSendToChannel {
 			options = append(options, slack.MsgOptionBroadcast())
 		}
 	}
-	_, threadTs, _, err := client.SendMessage(cfg.Channel, options...)
+	channelID, messageTs, _, err := client.SendMessage(cfg.Channel, options...)
 	if err != nil {
 		panic(err)
 	}
 
 	// If we already had a threadTs passed in (via env), keep using that one so
 	// that replies are threaded to the original thread. Replying to a reply
 	// doesn't work.
-	if cfg.ThreadTs != "" {
-		threadTs = cfg.ThreadTs
+	threadTs := cfg.ThreadTs
+	if threadTs == "" {
+		threadTs = messageTs
```

Comment:
what does this change do? is it just rearrangement so you can save the `messageTs` separately? it looks like the logic is the same as we had before but expressed in a slightly different way but I'm not sure

Reply:
>  is it just rearrangement so you can save the messageTs separately?

Yes. `client.SendMessage` now returns `messageTs`. `threadTs` is the timestamp of the first message of a thread, while messageTs is useful if you want to interact with a specific message within a thread. Before this change, we lost track of the timestamps of messages within a thread